### ISSUE TITLE
IDE0044 False Negative on Struct Field

### ISSUE THREAD

**Issue by nnpcYvIVl** [dotnet/roslyn#34115]
**Version Used**: 
VS 16.0.0 RC.2

**Steps to Reproduce**:
Load this code in the IDE
```csharp
namespace IDE0044_False_Negative
{
    class Class
    {
        int i;
        System.Drawing.Point p;
    }
}
```

**Expected Behavior**:
i and p should both be flagged for IDE0044: Make field readonly

**Actual Behavior**:
p is not flagged for IDE0044

**Comment by svick** [comment]
As far as I can tell, it is intentional that IDE0044 does not report fields of mutable value types like `Point`:

https://github.com/dotnet/roslyn/blob/087195e77cb9e47711289208cdcb07763a66dbbc/src/Features/Core/Portable/MakeFieldReadonly/MakeFieldReadonlyDiagnosticAnalyzer.cs#L104-L111

Why? Consider this code:

```c#
class C
{
    Point p;
    
    static void Main()
    {
        var c = new C();
        c.p.Offset(new Point(1,1));
        Console.WriteLine(c.p);
    }
}
```

Here, if you changed the field to `readonly`, the code would become incorrect, even though there is no obvious write to the field.

Because of this, I think the analyzer is behaving as designed and I also think that design is correct.

**Comment by sharwell** [comment]
@svick is correct here. `p` is not safe to be made `readonly` because it is a mutable value type.

**Comment by nnpcYvIVl** [comment]
@svick Thanks for the detailed explanation.
